StyleCop.Analyzers/StyleCop.Analyzers/Helpers/TriviaHelper.cs

```diff
@@ -311,6 +311,24 @@ internal static SyntaxToken WithoutLeadingBlankLines(this SyntaxToken token)
             return token.WithLeadingTrivia(newLeadingTrivia);
         }
 
+        private static bool HasBuiltinEndLine(this SyntaxTrivia trivia)
+        {
+            switch (trivia.Kind())
+            {
+            case SyntaxKind.SingleLineDocumentationCommentTrivia:
+            case SyntaxKind.PragmaWarningDirectiveTrivia:
```

Comment:
:exclamation: *All* directive trivia (i.e. the ones where `IsDirective` is true) contain a built-in end line. Some of them were left out here.